### PR TITLE
[App Service] `az webapp deleted restore`: Add new parameter `--target-app-svc-plan` to support setting app service plan for new azure web app

### PR DESCRIPTION
**Pull request by vkarasala**
**Related command**

az webapp deleted restore

**Description**

Changes are to improve the deleted restore experience and also to ensure the consistent behavior with PowerShell : https://learn.microsoft.com/en-us/azure/app-service/app-service-undelete
https://github.com/Azure/azure-powershell/blob/main/src/Websites/Websites/Cmdlets/BackupRestore/RestoreAzureDeletedWebApp.cs
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

**Comment by vkarasala** [comment]
Hi @zhoxing-ms, Is there way to re-run the pipeline(checks pipeline) without making a new commit? The last pipeline run is failed due to a test failure which is not related to my changes. Please advice. Thanks!

**Comment by zhoxing-ms** [comment]
> Running pylint on modules...
Pylint: PASSED
ERROR: ************* Module azure.cli.command_modules.appservice.custom
src/azure-cli/azure/cli/command_modules/appservice/custom.py:895:0: C0301: Line too long (121/120) (line-too-long)
src/azure-cli/azure/cli/command_modules/appservice/custom.py:898:0: C0301: Line too long (126/120) (line-too-long)
src/azure-cli/azure/cli/command_modules/appservice/custom.py:901:0: C0301: Line too long (123/120) (line-too-long)
src/azure-cli/azure/cli/command_modules/appservice/custom.py:895:0: C0301: Line too long (121/120) (line-too-long)
src/azure-cli/azure/cli/command_modules/appservice/custom.py:898:0: C0301: Line too long (126/120) (line-too-long)
src/azure-cli/azure/cli/command_modules/appservice/custom.py:901:0: C0301: Line too long (123/120) (line-too-long)

@vkarasala Could you please resolve the code style issue~

**Comment by vkarasala** [comment]
> > Running pylint on modules...
> > Pylint: PASSED
> > ERROR: ************* Module azure.cli.command_modules.appservice.custom
> > src/azure-cli/azure/cli/command_modules/appservice/custom.py:895:0: C0301: Line too long (121/120) (line-too-long)
> > src/azure-cli/azure/cli/command_modules/appservice/custom.py:898:0: C0301: Line too long (126/120) (line-too-long)
> > src/azure-cli/azure/cli/command_modules/appservice/custom.py:901:0: C0301: Line too long (123/120) (line-too-long)
> > src/azure-cli/azure/cli/command_modules/appservice/custom.py:895:0: C0301: Line too long (121/120) (line-too-long)
> > src/azure-cli/azure/cli/command_modules/appservice/custom.py:898:0: C0301: Line too long (126/120) (line-too-long)
> > src/azure-cli/azure/cli/command_modules/appservice/custom.py:901:0: C0301: Line too long (123/120) (line-too-long)
> 
> @vkarasala Could you please resolve the code style issue~

Thanks @zhoxing-ms , I have fixed it.

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -183,6 +183,7 @@ def load_arguments(self, _):
         c.argument('slot', options_list=['--slot', '-s'], help='slot to restore the deleted content to')
         c.argument('restore_content_only', action='store_true',
                    help='restore only deleted files without web app settings')
+        c.argument('target_app_svc_plan', options_list=['--target-app-svc-plan'], help='The app service plan for the new azure web app.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:186]: `--target-app-svc-plan` is the default option, please do not explicitly define it repeatedly
```suggestion
        c.argument('target_app_svc_plan', help='The app service plan for the new azure web app.')
```

Comment (vkarasala) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py]: Thanks for the suggestion @zhoxing-ms , Fixed it.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**